janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java

```diff
@@ -162,6 +163,86 @@
     public static final ConfigNamespace ES_INGEST_PIPELINES =
             new ConfigNamespace(ELASTICSEARCH_NS, "ingest-pipeline", "Ingest pipeline applicable to a store of an index.");
 
+    public static final ConfigNamespace SSL_NS =
+            new ConfigNamespace(ELASTICSEARCH_NS, "ssl", "Elasticsearch SSL configuration");
+
+    public static final ConfigOption<Boolean> SSL_ENABLED =
+            new ConfigOption<Boolean>(SSL_NS, "enabled",
+            "Controls use of the SSL connection to Elasticsearch.", ConfigOption.Type.LOCAL, false);
+
+    public static final ConfigOption<Boolean> SSL_DISABLE_HOSTNAME_VERIFICATION =
+            new ConfigOption<Boolean>(SSL_NS, "disable-hostname-verification",
+            "Disables the SSL hostname verification if set to true. Hostname verification is enabled by default.",
+            ConfigOption.Type.LOCAL, false);
+
+    public static final ConfigOption<Boolean> SSL_ALLOW_SELF_SIGNED_CERTIFICATES =
+            new ConfigOption<Boolean>(SSL_NS, "allow-self-signed-certificates",
+            "Controls the accepting of the self-signed SSL certificates.",
+            ConfigOption.Type.LOCAL, false);
```

Comment:
Can we combine these last two properties into one? Maybe `rejectUnauthorized` like in the [Javascript client](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/auth-reference.html#_https_ssl).

Reply:
I can probably imagine the usage of one of these options in some environments - for initial discovery or for on-going load balancing. In case of TCP-based load balancer (e.g. the one that does not terminate SSL/HTTP) I can imagine that disabling just the hostname validation while keeping the rest of the validation logic active may make sense - in some scenarios, of course.  

As long as it is only two safety switches, not 10, I personally prefer keeping them separate. As long as they are not hidden Java properties like in some other products. Each of these properties disables one particular check. To make your connections completely unprotected you need to use both.

Honestly, these days, with the free providers like letsencrypt, sslforfree etc I do not even know why one would bother setting a server with a self-signed certificate, even internally :) Especially considering that for ES you actually need to do extra work to get SSL support. Not like a product that only supports SSL and requires you to provide a valid certificate without offering plain HTTP option. SSL is no longer a luxury like 10 years ago :)

Reply:
Okay I was thinking the implementation would be the equivalent of the [TrustAllStrategy](https://issues.apache.org/jira/browse/HTTPCLIENT-1817) (added in 4.5.4) at which point the two properties would essentially represent the same thing and could be merged. But the slightly more restrictive TrustSelfSignedStrategy should still allow most if not all cases (it worked for my test case) so I'm okay with it.

As with most SSL client implementations supporting various even unrecommended configurations allows the final decisions to be made at the user level. There's documentation in the project (thanks to you) and elsewhere with recommendations and warnings on configuration best practices. By exposing the option (or options) we're giving users access to all the knobs in SSL configuration, including those we might recommend against using.